### PR TITLE
providers/google: Add support for encrypting a disk

### PR DESCRIPTION
**Pull request by danawillow**
This provides the ability to use raw customer-supplied encryption keys for disks created using the disk or instance resources.
https://cloud.google.com/compute/docs/disks/customer-supplied-encryption

### CODE REVIEW

File: builtin/providers/google/resource_compute_disk.go [hunk 1]
```diff
@@ -28,6 +28,17 @@ func resourceComputeDisk() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"disk_encryption_key_raw": &schema.Schema{
```
Comment (stack72) [inline builtin/providers/google/resource_compute_disk.go]: is this a sensitive key? If so, we should add Sensitive: true to the schema

File: builtin/providers/google/resource_compute_instance.go [hunk 2]
```diff
@@ -75,6 +75,17 @@ func resourceComputeInstance() *schema.Resource {
 							Type:     schema.TypeString,
 							Optional: true,
 						},
+
+						"disk_encryption_key_raw": &schema.Schema{
```
Comment (stack72) [inline builtin/providers/google/resource_compute_instance.go]: does this also need Sensitive: true ?

File: builtin/providers/google/resource_compute_instance.go [hunk 3]
```diff
@@ -770,6 +786,25 @@ func resourceComputeInstanceRead(d *schema.ResourceData, meta interface{}) error
...
+	for i, disk := range instance.Disks {
+		di := map[string]interface{}{
+			"disk":                    d.Get(fmt.Sprintf("disk.%d.disk", i)),
+			"image":                   d.Get(fmt.Sprintf("disk.%d.image", i)),
+			"type":                    d.Get(fmt.Sprintf("disk.%d.type", i)),
+			"scratch":                 d.Get(fmt.Sprintf("disk.%d.scratch", i)),
+			"auto_delete":             d.Get(fmt.Sprintf("disk.%d.auto_delete", i)),
+			"size":                    d.Get(fmt.Sprintf("disk.%d.size", i)),
+			"device_name":             d.Get(fmt.Sprintf("disk.%d.device_name", i)),
+			"disk_encryption_key_raw": d.Get(fmt.Sprintf("disk.%d.disk_encryption_key_raw", i)),
```
Comment (stack72) [inline builtin/providers/google/resource_compute_instance.go]: Do we want to add the raw encryption key to the state?

Comment (stack72) [inline builtin/providers/google/resource_compute_instance.go]: FYI, this is not happening in compute_disk

File: builtin/providers/google/resource_compute_disk.go (hunk 1 above)
Comment (danawillow) [inline builtin/providers/google/resource_compute_disk.go]: Done

File: builtin/providers/google/resource_compute_instance.go (hunk 2 above)
Comment (danawillow) [inline builtin/providers/google/resource_compute_instance.go]: Done

File: builtin/providers/google/resource_compute_instance.go (hunk 3 above)
Comment (danawillow) [inline builtin/providers/google/resource_compute_instance.go]: I think you're right. Removed.

**Comment by stack72** [comment]
this looks great now @danawillow :)

```
% make testacc TEST=./builtin/providers/google TESTARGS='-run=TestAccComputeDisk_'                                                        ✭
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/01/18 13:46:38 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/google -v -run=TestAccComputeDisk_ -timeout 120m
=== RUN   TestAccComputeDisk_basic
--- PASS: TestAccComputeDisk_basic (29.67s)
=== RUN   TestAccComputeDisk_encryption
--- PASS: TestAccComputeDisk_encryption (29.65s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/google	59.335s
```

**Pull request merged**